### PR TITLE
More netwatch metrics

### PR DESCRIPTION
**Pull request by hsmade**
Added metrics for the different types: icmp, tcp, http, https.
Added the type label to the metrics, as it's possible to have more monitors with the same name (ip), but different types.

```
mktxp_netwatch_info{comment="",host="1.2.3.4",interval="1m",name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",since="2023-11-02 22:31:49",status="1",timeout=""} 1.0
mktxp_netwatch_info{comment="",host="192.168.88.1",interval="",name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",since="2023-11-02 23:21:48",status="0",timeout=""} 1.0
mktxp_netwatch_info{comment="",host="192.168.88.1",interval="",name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",since="2023-11-02 23:21:58",status="0",timeout=""} 1.0
mktxp_netwatch_info{comment="",host="192.168.88.1",interval="",name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",since="2023-11-02 23:22:13",status="1",timeout=""} 1.0
mktxp_netwatch_status{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 1.0
mktxp_netwatch_status{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="http-get"} 0.0
mktxp_netwatch_status{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="https-get"} 0.0
mktxp_netwatch_status{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="tcp-conn"} 1.0
mktxp_netwatch_done_tests{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 58.0
mktxp_netwatch_done_tests{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="http-get"} 45.0
mktxp_netwatch_done_tests{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="https-get"} 44.0
mktxp_netwatch_done_tests{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="tcp-conn"} 43.0
mktxp_netwatch_failed_tests{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 0.0
mktxp_netwatch_failed_tests{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="http-get"} 45.0
mktxp_netwatch_failed_tests{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="https-get"} 44.0
mktxp_netwatch_failed_tests{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="tcp-conn"} 0.0
mktxp_netwatch_icmp_loss_count{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 0.0
mktxp_netwatch_icmp_loss_percent{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 0.0
mktxp_netwatch_icmp_rtt_avg_ms{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 1.094
mktxp_netwatch_icmp_rtt_min_ms{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 1.004
mktxp_netwatch_icmp_rtt_max_ms{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 1.141
mktxp_netwatch_icmp_rtt_jitter_ms{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 0.137
mktxp_netwatch_icmp_rtt_stdev_ms{name="1.2.3.4",routerboard_address="192.168.88.1",routerboard_name="test",type="icmp"} 0.051
mktxp_netwatch_tcp_connect_time_ms{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="tcp-conn"} 208.214
mktxp_netwatch_http_status_code{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="http-get"} 301.0
mktxp_netwatch_http_status_code{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="https-get"} 404.0
mktxp_netwatch_http_resp_time{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="http-get"} 255.43
mktxp_netwatch_http_resp_time{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="https-get"} 275.809
mktxp_netwatch_tcp_connect_time_ms{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="http-get"} 253.508
mktxp_netwatch_tcp_connect_time_ms{name="192.168.88.1",routerboard_address="192.168.88.1",routerboard_name="test",type="https-get"} 274.002

```

### CODE REVIEW

File: mktxp/collector/netwatch_collector.py [hunk 1]
```diff
@@ -25,11 +26,61 @@ def collect(router_entry):
...
+                'rtt_jitter': lambda value: NetwatchCollector.parse_timedelta(value).microseconds/1000,
+                'rtt_stdev': lambda value: NetwatchCollector.parse_timedelta(value).microseconds/1000,
+                'tcp_connect_time': lambda value: NetwatchCollector.parse_timedelta(value).microseconds/1000,
+                'http_resp_time': lambda value: NetwatchCollector.parse_timedelta(value).microseconds/1000,
+            }[label](value)
+        except KeyError:
+            return value
+
+    @staticmethod
+    def parse_timedelta(time):
```
Comment (hsmade) [inline mktxp/collector/netwatch_collector.py:83]: The one in `BaseOutputProcessor` breaks because the `m` for minutes matches the `ms` in the value

**Comment by hsmade** [comment]
Implements https://github.com/akpw/mktxp/issues/81

**Comment by byjosez** [comment]
You can add the number of registered and active pppoe sessions?

**Comment by frittentheke** [comment]
This is really a cool addition to have the built-in netwatch health checks exported as metrics!
@byjosez Why not create a new PR for PPPoE metrics? I'd really love to see this one here get done and merged.

**Comment by akpw** [comment]
@hsmade I ended up transforming this into a [separate commit](https://github.com/akpw/mktxp/commit/e473d1ee263bf3dc35b5e60b5cfef552460fd21e), fixing the mentioned BaseOutputProcessor issue & aligning with recent mktxp processing changes

**Pull request closed without merging**